### ISSUE TITLE
Unable to download/fetch Gitlab Packages

### ISSUE THREAD

**Issue by kennyballou** [nix-community/emacs-overlay#128]
I noticed `geiser` was marked broken when I tried to update to the latest commit in the overlay.

Looking over the [latest commit][0], it appears to be the Gitlab fetcher is not finding packages?  Manually trying the URL's generated in the output give me 404's, not 503's as the output indicates.

Maybe this isn't an issue for the overlay, but within fetcher in nixpkgs?

[0]: https://github.com/nix-community/emacs-overlay/commit/cab95bd5e977ca4dec63e98b4ce103955fce45a6

**Comment by terlar** [comment]
This is not an issue with the fetcher in nixpkgs, but related to the update-time. I fixed this particular package by adding the following override (https://github.com/terlar/emacs-config/blob/master/overrides.nix#L22-L33):
```
  geiser = epkgs.melpaPackages.geiser.overrideAttrs (attrs: {
    src = fetchFromGitLab {
      owner = "emacs-geiser";
      repo = "geiser";
      rev = "d7ba81b402787e3315b40f60952f95816a1cf99c";
      sha256 = "sha256-we0un+EYB2ByA57g+TVMuP21HBmXRaTLt5JC4QWjGAE=";
    };

    meta = attrs.meta // {
      broken = false;
    };
  });
```

**Comment by talyz** [comment]
This looks like a bug in `emacs2nix`. Reported in https://github.com/ttuegel/emacs2nix/issues/56.

**Comment by kennyballou** [comment]
Now that ttuegel/emacs2nix#57 is merged, do we wait for the next update to resolve GitLab packages, or is there some related work in the repository that is necessary to use the new version of emacs2nix?

**Comment by kennyballou** [comment]
> Now that [ttuegel/emacs2nix#57](https://github.com/ttuegel/emacs2nix/pull/57) is merged, do we wait for the next update to resolve GitLab packages, or is there some related work in the repository that is necessary to use the new version of emacs2nix?

Based on the contents in bb8d4e9862410bfd1c4dfca5193248edb634be43, there's work to be done.

**Comment by kennyballou** [comment]
This has been interesting to unravel, I'm not a hundred percent I've completely untangled how this is all coming together.  But I believe this issue will be resolved when/if NixOS/nixpkgs#120354 merges.

**Comment by kennyballou** [comment]
NixOS/nixpkgs#120354 has merged.  I will wait for the next melpa update to be committed and will close this issue after verifying the expected results.

**Comment by kennyballou** [comment]
As of b717ab8d43ecabd27aa884c278a4edd1caa642db, this looks to be resolved.

**Comment by chvp** [comment]
I'm still having issues with `modus-themes`, but maybe that's due to #125? The version it tries to build is `20210503.1941`, but the version on melpa is `20210504.902`.

**Comment by matthew-piziak** [comment]
@chvp I had trouble with modus themes myself. I ended up loading them my `.emacs` instead, using this use-package form:
```elisp
  (use-package modus-themes
    :straight (modus-themes :type git :host gitlab :repo "protesilaos/modus-themes")
    :init
    (modus-themes-load-themes)
    :hook (after-init . modus-themes-load-vivendi))
```

**Comment by kennyballou** [comment]
@chvp: I think you're right that it's related to #125; I also saw this issue with the `modus-themes`.  Looking at the commits on the project, a new commit was added a few hours after the auto update finished, likely triggering the unstable URL path.

That said, the new head, 0e85ce568b31f4bca66c4edac93bd51a166382ac, looks to have the right commit and builds for modus-themes. (I also noticed that melpa-stable version of `modus-themes` doesn't build either.)
